### PR TITLE
CPP: Add syntax examples to QLDoc in Enum.qll.

### PR DESCRIPTION
**Pull request by geoffw0**
For https://jira.semmle.com/browse/CPP-407.

I also extended a test.

### CODE REVIEW

File: cpp/ql/src/semmle/code/cpp/Enum.qll [hunk 1]
```diff
@@ -2,7 +2,13 @@ import semmle.code.cpp.Type
 private import semmle.code.cpp.internal.ResolveClass
 
 /**
- * A C/C++ enum [N4140 7.2].
+ * A C/C++ enum [N4140 7.2]. For example, the type `MyEnum` in:
+ * ```
+ * enum MyEnum
```
Comment (jbj) [inline cpp/ql/src/semmle/code/cpp/Enum.qll]: It doesn't have to be in the example, but I think it's worth stating that this class includes C++11 scoped enums (`ScopedEnum` in QL).

File: cpp/ql/src/semmle/code/cpp/Enum.qll [hunk 2]
```diff
@@ -46,7 +52,17 @@ class Enum extends UserType, IntegralOrEnumType {
...
 
 /**
- * A C++ enum that is directly enclosed by a function.
+ * A C++ enum that is directly enclosed by a function. For example, the type
+ * `MyLocalEnum` in:
+ * ```
+ * void myFunction()
+ * {
+ *   enum MyLocalEnum
+ *   {
```
Comment (jbj) [inline cpp/ql/src/semmle/code/cpp/Enum.qll]: Brace placement varies over this file. I prefer the style we settled on for `Class.qll`, so this example would become

```
void myFunction() {
  enum MyLocalEnum {
     MyLocalEnumConstant
  };
}
```

File: cpp/ql/src/semmle/code/cpp/Enum.qll [hunk 3]
```diff
@@ -46,7 +52,17 @@ class Enum extends UserType, IntegralOrEnumType {
 }
 
 /**
- * A C++ enum that is directly enclosed by a function.
+ * A C++ enum that is directly enclosed by a function. For example, the type
```
Comment (jbj) [inline cpp/ql/src/semmle/code/cpp/Enum.qll]: "C++" -> "C/C++".

File: cpp/ql/src/semmle/code/cpp/Enum.qll [hunk 4]
```diff
@@ -57,7 +73,18 @@ class LocalEnum extends Enum {
 }
 
 /**
- * A C++ enum that is declared within a class.
+ * A C++ enum that is declared within a class. For example, the type
```
Comment (jbj) [inline cpp/ql/src/semmle/code/cpp/Enum.qll]: "C++" -> "C/C++".

Comment (geoffw0) [reply cpp/ql/src/semmle/code/cpp/Enum.qll]: Because it could be enclosed in a struct?

Done.

File: cpp/ql/src/semmle/code/cpp/Enum.qll (hunk 3 above)
Comment (geoffw0) [reply cpp/ql/src/semmle/code/cpp/Enum.qll]: Done.

File: cpp/ql/src/semmle/code/cpp/Enum.qll (hunk 2 above)
Comment (geoffw0) [reply cpp/ql/src/semmle/code/cpp/Enum.qll]: Fixed.

File: cpp/ql/src/semmle/code/cpp/Enum.qll (hunk 1 above)
Comment (geoffw0) [reply cpp/ql/src/semmle/code/cpp/Enum.qll]: Done.

**Comment by jbj** [comment]
Now you've changed the brace placement in this PR (and others, I think) to put the opening brace on its own line, but in #1570 we agreed on placing the opening brace at the end of the declaration line, and this is what got merged into master: https://github.com/Semmle/ql/blob/3b0ddf54bc0e5eaa862c466ee27e98d5d95cf3c3/cpp/ql/src/semmle/code/cpp/Class.qll#L14-L26.

File: cpp/ql/src/semmle/code/cpp/Enum.qll (hunk 4 above)
Comment (jbj) [reply cpp/ql/src/semmle/code/cpp/Enum.qll]: Yes.

**Comment by geoffw0** [comment]
> we agreed on placing the opening brace at the end of the declaration line

Ah, sorry, I will fix it up (on all three affected PRs).

**Comment by geoffw0** [comment]
Done.

**Comment by geoffw0** [comment]
> the tests are failing

I accidentally committed a change to the test I'd made while debugging.  Now fixed.

**Pull request merged**